spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/PrefixContext.java

```diff
@@ -16,15 +16,16 @@
 
 package org.springframework.cloud.kubernetes.commons.config;
 
-import java.util.Map;
-import java.util.Set;
+import java.util.function.Function;
 
 /**
  * A holder for data needed to compute prefix based properties, in case of a secret or
  * config map.
  *
  * @author wind57
  */
-public final record PrefixContext(Map<String, Object> data, String prefix, String namespace,
-		Set<String> propertySourceNames) {
+public final record PrefixContext(MultipleSourcesContainer data, Function<String, String> prefix, String namespace) {
```

Comment:
this is a breaking change and unfortunately we can't merge breaking changes in `main`. This only happens on "next major" releases. afaik: next one is at the end of 2025. Ryan can confirm (I'm not a vmware employee), but it seems this would be a NOGO, for the time being at least. What I do is maintain some PRs that are targeted towards next major, but this is not easy, to say the least...

Reply:
I have not looked at anything else in the code tbh, will wait for Ryan's input

Reply:
We had discussed in the original issue that there would be breaking changes and it would need to [wait for a major](https://github.com/spring-cloud/spring-cloud-kubernetes/issues/1757#issuecomment-2416830110).